internal/reader/reader.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2022-present Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package reader
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+
+	"github.com/omec-project/metricfunc/config"
+	"github.com/omec-project/metricfunc/internal/metricdata"
+	"github.com/omec-project/metricfunc/pkg/metricinfo"
+	"github.com/segmentio/kafka-go"
+)
+
+func StartKafkaReader(cfg *config.Configuration) {
+
+	//Start Kafka Event Reader
+	for _, nfStream := range cfg.NfStreams {
+		r := kafka.NewReader(kafka.ReaderConfig{
+			Brokers: nfStream.Urls,
+			Topic:   nfStream.Topic.TopicName,
+			GroupID: nfStream.Topic.TopicGroups,
+		})
+
+		go reader(r)
+	}
+}
+
+func getSourceNfType(r *kafka.Reader) metricinfo.NfType {
+
+	topic := r.Config().Topic
+	switch string(topic[len(topic)-3:]) {
```

Comment:
Sounds error prone. We should have better way of doing this.

Reply:
right, kept original topic name as per config name("sdcore-data-source-smf"/"sdcore-data-source-amf")